heft-plugins/heft-isolated-typescript-transpile-plugin/src/SwcIsolatedTranspilePlugin.ts

```diff
@@ -0,0 +1,384 @@
+      rootMode: 'root',
+      configFile: false,
+      swcrc: false,
+      minify: false,
+
+      inputSourceMap: false,
+      sourceRoot,
+      isModule: true,
+
+      module: moduleConfig,
+      jsc: {
+        target,
+        externalHelpers: tsConfigOptions.importHelpers,
+        parser,
+        transform: {
+          legacyDecorator: experimentalDecorators,
+          react,
+          useDefineForClassFields,
+          // This property is not included in the types, but is what makes swc-jest work
+          // @ts-ignore
+          hidden: {
+            jest: format === 'commonjs'
+          }
+        }
+      }
+    };
+
+    if (getSwcOptionsHook.isUsed()) {
+      options = getSwcOptionsHook.call(options, formatOverride, targetOverride);
+    }
+
+    logger.terminal.writeVerboseLine(`Transpile options: ${JSON.stringify(options, undefined, 2)}}`);
+    logger.terminal.writeDebugLine(`Transpile options: ${options}`);
+
+    const tsOptions: string = JSON.stringify(options);
+    const tsOptionsBuffer: Buffer = Buffer.from(new SharedArrayBuffer(Buffer.byteLength(tsOptions, 'utf8')));
+    tsOptionsBuffer.write(tsOptions);
+    parser.tsx = true;
+    const tsxOptions: string = JSON.stringify(options);
+    const tsxOptionsBuffer: Buffer = Buffer.from(
+      new SharedArrayBuffer(Buffer.byteLength(tsxOptions, 'utf8'))
+    );
+    tsxOptionsBuffer.write(tsxOptions);
+
+    return {
+      ts: tsOptionsBuffer,
+      tsx: tsxOptionsBuffer
+    };
+  }
+
+  const outputOptions: Map<string, IOptionsBufferByExtension> = new Map();
+  for (const emitKind of emitKinds) {
+    outputOptions.set(emitKind.outDir, getOptionsBufferForFormatAndTarget(emitKind));
+  }
+
+  const tasks: ITransformTask[] = [];
+
+  for (const srcFilePath of sourceFilePaths) {
+    const rootPrefixLength: number | undefined = rootDirsPaths.findChildPath(srcFilePath);
+
+    if (rootPrefixLength === undefined) {
+      throw new Error(`Could not determine root prefix for ${srcFilePath}}`);
+    }
+
+    const relativeSrcFilePath: string = srcFilePath.slice(rootPrefixLength);
+    const extensionIndex: number = relativeSrcFilePath.lastIndexOf('.');
+    const tsx: boolean = endsWithCharacterX(relativeSrcFilePath);
+
+    const relativeJsFilePath: string = `${relativeSrcFilePath.slice(0, extensionIndex)}.js`;
+    for (const [outputPrefix, optionsByExtension] of outputOptions) {
+      const jsFilePath: string = `${outputPrefix}${relativeJsFilePath}`;
+      const mapFilePath: string | undefined = externalSourceMaps ? `${jsFilePath}.map` : undefined;
+
+      const item: ITransformTask = {
+        srcFilePath,
+        relativeSrcFilePath,
+        options: tsx ? optionsByExtension.tsx : optionsByExtension.ts,
+        jsFilePath,
+        mapFilePath
+      };
+
+      tasks.push(item);
+    }
+  }
+
+  logger.terminal.writeLine(`Transpiling ${tasks.length} files...`);
+
+  const result: IWorkerResult = await new Promise((resolve, reject) => {
```

Comment:
Due to https://github.com/rust-lang/rust/issues/91979 using `worker_threads` is not recommended for swc & napi-rs. 

Will message you the ADO implementation PR for migrating this to child_process.